apps/service_providers/tests/test_runnables_cancellation.py

```diff
@@ -87,6 +88,13 @@ def _stream(*args, **kwargs):
 
 def _get_mocked_history_recording(session, runnable_cls):
     adapter = ChatAdapter.for_experiment(session.experiment, session)
-    runnable = runnable_cls(adapter=adapter, check_every_ms=0)
+    history_manager = ExperimentHistoryManager.for_llm_chat(
+        session=session,
+        experiment=session.experiment,
+        max_token_limit=session.experiment.max_token_limit,
+        chat_model=session.experiment.get_chat_model(),
+        trace_service=session.experiment.trace_service,
+    )
+    runnable = runnable_cls(adapter=adapter, history_manager=history_manager, check_every_ms=0)
```

Comment:
Would you mind elaborating? Are you saying that experiment is passed to ChatAdapter and ExperimentHistoryManager?

Reply:
I see. This is by design, since the session's experiment will not always be the one being passed to the history manager. Let's keep it as-is. I have plans to maybe refactor this down the line